File: web-service/api-project/src/main/java/io/ehdev/conrad/service/api/service/model/project/CreateProjectModel.java
package io.ehdev.conrad.service.api.service.model.project;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import org.springframework.hateoas.ResourceSupport;

public class CreateProjectModel extends ResourceSupport {

    @JsonPropertyDescription("Name of the project created")
    @JsonProperty("name")
    String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: api/rest-model/src/main/kotlin/io/ehdev/conrad/model/project/CreateProjectModel.kt
package io.ehdev.conrad.model.project

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyDescription
import io.ehdev.conrad.model.ResourceLink
import io.ehdev.conrad.model.ResourceSupport

data class CreateProjectModel(@JsonPropertyDescription("Name of the project created")
                              @JsonProperty("name")
                              val name: String,

                              @JsonProperty("links")
                              override val links: List<ResourceLink>) : ResourceSupport {

}
